platform/linuxbsd/os_linuxbsd.cpp

```diff
@@ -123,17 +123,32 @@ String OS_LinuxBSD::get_name() const {
 
 Error OS_LinuxBSD::shell_open(String p_uri) {
 	Error ok;
+	int errCode;
 	List<String> args;
 	args.push_back(p_uri);
-	ok = execute("xdg-open", args, false);
-	if (ok == OK) {
+
+	// Agnostic
+	ok = execute("xdg-open", args, true, nullptr, nullptr, &errCode);
+	if (ok == OK && !errCode) {
```

Comment:
I'm not sure about this, if there's a failed error code returns by the process, most likely the path we're trying to open cannot be opened, and going through attempt all `-open` methods under the sun is probably not a great idea (it will likely spam a lot of duplicate errors about e.g. path not found).

Reply:
Apparently `xdg-open`, `gvfs-open` and `gio open` return exit code 2 if the specified file/path doesn't exist *(and while `kde-open` doesn't have a exit code for that AFAIK, it opens a window with the error)*, so I'll add that into account.